web/modules/custom/dbcdk_community/src/Plugin/Block/ProfileBlock.php

```diff
@@ -0,0 +1,247 @@
+use Drupal\Core\Plugin\ContainerFactoryPluginInterface;
+use Drupal\Core\Block\BlockBase;
+use Drupal\Core\Url;
+use Drupal\Component\Utility\Xss;
+
+/**
+ * Provides a 'ProfileBlock' block.
+ *
+ * This block provides a display of a single Community Service Profile.
+ *
+ * @Block(
+ *   id = "dbcdk_community_profile_block",
+ *   admin_label = @Translation("DBCDK Community Profile"),
+ *   context = {
+ *     "username" = @ContextDefinition("string")
+ *   }
+ * )
+ */
+class ProfileBlock extends BlockBase implements ContainerFactoryPluginInterface {
+
+  use CommunityTraits;
+
+  /**
+   * The DBCDK Community Service Profile API.
+   *
+   * @var ProfileApi $profileApi
+   */
+  protected $profileApi;
+
+  /**
+   * Creates a Profiles Block instance.
+   *
+   * @param array $configuration
+   *   A configuration array containing information about the plugin instance.
+   * @param string $plugin_id
+   *   The plugin_id for the plugin instance.
+   * @param mixed $plugin_definition
+   *   The plugin implementation definition.
+   * @param \DBCDK\CommunityServices\Api\ProfileApi $profile_api
+   *   The DBCDK Community Service Profile API.
+   */
+  public function __construct(array $configuration, $plugin_id, $plugin_definition, ProfileApi $profile_api) {
+    parent::__construct($configuration, $plugin_id, $plugin_definition);
+    $this->profileApi = $profile_api;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container, array $configuration, $plugin_id, $plugin_definition) {
+    return new static(
+      $configuration,
+      $plugin_id,
+      $plugin_definition,
+      $container->get('dbcdk_community.api.profile')
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function build() {
+    // Tries to fetch a profile from the Community Service.
+    // We catch any API Exceptions and stop the rest of the code from executing
+    // so we don't get fatal errors the will cause a "status code 500" but
+    // rather displays an empty table.
+    $profile = NULL;
+    $username = $this->getContext('username')->getContextData()->getValue();
+    try {
+      $filter = [
+        'limit' => 1,
+        'where' => [
+          'username' => $username,
+        ],
+      ];
+
+      // Since we have a limit of 1 result, we simply select that one result
+      // from the results array instead of looping through it.
+      $profile = $this->profileApi->profileFind(json_encode($filter))[0];
+    }
+    catch (\Exception $e) {
+      $this->handleException($e);
+    }
+
+    // Defensive coding to make sure we don't break anything if the API returns
+    // "NULL" or something similar instead of an array of profiles.
+    // Defensive coding to make sure we don't cause a fatal error by processing
+    // the results from the request as a profile, without it being a profile
+    // object - The results could return NULL.
+    if (!empty($profile) && $profile instanceof Profile) {
```

Comment:
The `empty` check was also to make sure the variable was set because the profile variable was only set inside the the `try {}` - but this is no longer the case. Good catch!